script/schema_doc.py

```
@@ -0,0 +1,910 @@
+        index = 1
+        while lines[index] != "---":
+            if lines[index].startswith("title: "):
+                md_docs[md_file]["title"] = lines[index][len("title:  ") : -1]
+            index += 1
+        return index + 1
+    return 0
+
+
+def open_file_lines(file):
+    if os.path.exists(file):
+        file_f = open(file, "r", encoding="utf-8-sig")
```

Ruff (0.12.2)

Use a context manager for opening files

(SIM115)